test/ex_machina/ecto_test.exs

```diff
@@ -44,18 +45,38 @@ defmodule ExMachina.EctoTest do
     }
   end
 
-  test "params_for/2 raises when passed a map" do
-    assert_raise ArgumentError, fn ->
-      TestFactory.params_for(:user_map)
-    end
-  end
-
   test "params_for/2 removes fields with nil values" do
     assert TestFactory.params_for(:user, admin: nil) == %{
-      name: "John Doe"
+      name: "John Doe",
+      articles: [],
     }
   end
 
+  test "params_for/2 recursively strips has_many values" do
+    article = TestFactory.build(:article)
+
+    user_params = TestFactory.params_for(:user, articles: [article])
+
+    assert user_params[:articles] == [%{
```

Comment:
Yes, I see your point, thanks 👍 !

Just one little thing - the test description mentions _strips has_many values_ (which sounds like _deleting_ them), while assertion here seems to prove opposite. Unless it's just me - would rewording the description to _preserves has_many values/associations_ make it clearer?